File: sprint6/JS/lesson5code1.js
let color = ["white", "white", ...];
let time = 0;
let entry = [null, null, ...];
let leave = [null, null, ...];

function DFS(v) {
    time += 1;
    entry[v] = time;
    color[v] = "gray";
    for (let w of getOutgoingEdges(v)) {
        if (color[w] === "white") {
            DFS(w);
        }
    }
    time += 1;
    leave[v] = time;
    color[v] = "black";
}

File: sprint6/Java/lesson5code1.java
List<String> color = new ArrayList<>(Arrays.asList("white", "white", ...));
int time = 0;
List<Integer> entry = new ArrayList<>(Arrays.asList(null, null, ...));
List<Integer> leave = new ArrayList<>(Arrays.asList(null, null, ...));

void DFS(int v) {
    time += 1;
    entry.set(v, time);
    color.set(v, "gray");
    for (int w : getOutgoingEdges(v)) {
        if (color.get(w).equals("white")) {
            DFS(w);
        }
    }
    time += 1;
    leave.set(v, time);
    color.set(v, "black");
}
